### PR TITLE
fix: updated Rights Tab Config w/ form field config

### PR DESCRIPTION
**Pull request by suhail-ansari**
## Description of this change

> updated Rights Tab Config w/ form field config

## Why is this change being made?
- [ ] Chore (non-functional changes)
- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How was this tested? How can the reviewer verify your testing?

Screen Shot 2023-03-16 at 3 08 29 PM

## Related issues

[KD-6420](https://ketch-com.atlassian.net/browse/KD-6420)

## Checklist

- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] I have evaluated the security impact of this change, and [OWASP Secure Coding Practices](https://owasp.org/www-project-secure-coding-practices-quick-reference-guide/migrated_content#) have been observed.
- [x] I have informed stakeholders of my changes.

[KD-6420]: https://ketch-com.atlassian.net/browse/KD-6420?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by keepforever — APPROVED** [review]

File: src/index.ts [hunk 1]
```diff
@@ -671,6 +671,102 @@ export interface JIT {
...
+ * FormFieldConfig
+ */
+export interface FormFieldConfig {
+  formFieldId?: string
+  required?: boolean
+  label?: string
+  hint?: string
+  category?: FormFieldCategory
+  type?: FormFieldType
+  Variant?: FormFieldVariant
```
Comment (sethyates) [inline src/index.ts:693]: case

File: src/index.ts [hunk 2]
```diff
@@ -671,6 +671,102 @@ export interface JIT {
...
+  options?: FormFieldDropdownOption[]
+}
+
+/**
+ * FormFieldDropdownOption
+ */
+export type FormFieldDropdownOption = {
+  label?: string
+  value?: string
+  translations?: { [key: string]: string }
```
Comment (sethyates) [inline src/index.ts:707]: How does this work?

File: src/index.ts [hunk 3]
```diff
@@ -671,6 +671,102 @@ export interface JIT {
...
+export interface FormFieldIdentitySpace {
+  id?: string
+  name?: string
+  code?: string
+}
+
+/**
+ * FormFieldConfig
+ */
+export interface FormFieldConfig {
```
Comment (sethyates) [inline src/index.ts:686]: Why isn't this just `FormField`?

File: src/index.ts [hunk 4]
```diff
@@ -671,6 +671,102 @@ export interface JIT {
...
+  formFieldId?: string
```
Comment (sethyates) [inline src/index.ts:687]: `id`?

File: src/index.ts [hunk 5]
```diff
@@ -684,6 +780,25 @@ export interface RightsTab {
...
+  /**
+   * Personal Details Section
+   */
+  personalDetailsTitle?: string
+  showPersonalDetailsTitle?: boolean
+
+  /**
+   * Form Fields Config for Custom Rights Form
+   */
+  inputConfigs?: FormFieldConfig[]
```
Comment (sethyates) [inline src/index.ts:801]: `formFields`?

File: src/index.ts [hunk 6]
```diff
@@ -671,6 +671,102 @@ export interface JIT {
   extensions?: { [key: string]: string }
 }
 
+/**
+ * FormFieldIdentitySpace
+ */
+export interface FormFieldIdentitySpace {
+  id?: string
+  name?: string
+  code?: string
```
Comment (sethyates) [inline src/index.ts:680]: this is a very different setup to normal identity spaces. can you please explain what is going on here?

File: src/index.ts (hunk 2 above)
Comment (suhail-ansari) [reply src/index.ts:707]: pretty much like `<option value={value}>{label}</option>`
translations are `langCode: labelTranslation`

File: src/index.ts (hunk 6 above)
Comment (suhail-ansari) [reply src/index.ts:680]: form field can be associated w/ formField, we just associate formField w/ identity using `ID` as foreign to pull in that info. the formField will use the `formField.identitySpace.code` to add identity to the the right-invocation payload

File: src/index.ts (hunk 1 above)
Comment (suhail-ansari) [reply src/index.ts]: fixed ✅

File: src/index.ts (hunk 3 above)
Comment (suhail-ansari) [reply src/index.ts]: fixed ✅

File: src/index.ts (hunk 4 above)
Comment (suhail-ansari) [reply src/index.ts]: fixed ✅

File: src/index.ts (hunk 5 above)
Comment (suhail-ansari) [reply src/index.ts]: fixed ✅

File: src/index.ts (hunk 6 above)
Comment (sethyates) [reply src/index.ts:680]: I still don't understand your message. What is `id` and `name` for?

File: src/index.ts [hunk 7]
```diff
@@ -671,6 +671,102 @@ export interface JIT {
...
+}
+
+/**
+ * FormFieldConfig
```
Comment (sethyates) [inline src/index.ts:684]: Comment

File: src/index.ts (hunk 2 above)
Comment (sethyates) [reply src/index.ts:707]: If this is returned in the experience, then the `value` should already be translated. The `translations` like this are only used in supercargo.

File: src/index.ts (hunk 7 above)
Comment (suhail-ansari) [reply src/index.ts]: fixed ✅

**Review by sethyates — APPROVED** [review]

**Review by sethyates — APPROVED** [review]

**Pull request merged**